data/lua/core.lua

```diff
@@ -202,26 +202,32 @@ function wesnoth.deprecate_api(elem_name, replacement, level, version, elem, det
 			return elem(...)
 		end
 	elseif type(elem) == "table" then
-		local mt = {
-			__index = function(self, key)
-				if not msg_shown then
-					msg_shown = true
-					wesnoth.deprecated_message(elem_name, level, version, message)
-				end
-				return elem[key]
-			end,
-			__newindex = function(self, key, val)
-				if not msg_shown then
-					msg_shown = true
-					wesnoth.deprecated_message(elem_name, level, version, message)
-				end
-				elem[key] = val
-			end,
-		}
 		-- Don't clobber the old metatable.
 		local old_mt = getmetatable(elem)
-		if type(old_mt) == "table" then
-			setmetatable(mt, old_mt)
+		if old_mt == nil then old_mt = {} end
```

Comment:
This line could be merged with the previous one as `local old_mt = getmetatable(elem) or {}`.

Reply:
Fixed